cpp/oneapi/dal/algo/decision_forest/common.hpp

```diff
@@ -495,6 +581,15 @@ class model : public base {
 } // namespace v1
 
 using v1::descriptor;
+using v1::node_descriptor;
+using v1::split_node_descriptor;
+using v1::leaf_node_descriptor;
+using v1::is_leaf_node_descriptor;
+using v1::is_leaf_node_descriptor_v;
+using v1::is_split_node_descriptor;
+using v1::is_split_node_descriptor_v;
 using v1::model;
 
 } // namespace oneapi::dal::decision_forest
+
+#include "oneapi/dal/algo/decision_forest/detail/tree_node_visitor.hpp"
```

Comment:
Not sure if it's good idea...

Headers must be self contained. This is part of our guidelines and C++ core guidelines https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#sf11-header-files-should-be-self-contained

Possible solutions:
- Introduce forward declarations in `tree_node_visitor.hpp` (probably it will not work since forward declarations are not enough)
- Declare node descriptors in separate file such as `decision_forest/tree_nodes.hpp`, probably forward declaration to task will be necessary

Reply:
rewrote it